### DISCUSSION TITLE
Custom channels?

### DISCUSSION THREAD

**Discussion by Enchiridion** [Pierre-Lannoy/wp-decalog-sdk#2] [Q&A]
Hi,
Nice plugin!
Do you have any examples of how to create and use custom channels?
Also will it be possible in the future to save extra data besides 'code'?  I'm currently using Monolog and am used to saving additional context data with the messages.
Is it possible to disable the nonsense table names when using the WordPress handler?  Names like `wp_decalog_cb705408b664406485deb93f534dbd30` mean nothing.  If a unique name is required, I'd like to specify it myself.
Thanks!

**Comment by Pierre-Lannoy** [comment]
Hello @Enchiridion … And thanks for your kind words.

I will try to answer your questions ;)

**Custom Channels**: there's no way to use different channels. DecaLog uses a [specific definition](https://github.com/Pierre-Lannoy/wp-decalog/blob/master/LOGGING.md#anatomy-of-an-event) for channels and, based on this definition, I think the whole perimeter is covered. Nevertheless, if you have in mind a channel not covered, don't hesitate to tell me.

**Extra context**: yes ! It is already in my backlog but not yet prioritised. So don't expect it before summer 2022.

**Table name**: the WP internal loggers (those with a table in WP) are not designed to be accessed by other extensions than DecaLog. It's why table names are managed by DecaLog itself and can't be customised. I'm very curious about the use case needing direct accesses to these tables, could you tell me more?

**Comment by Enchiridion** [comment]
Custom Channels: My sites often have integrations with 3rd party services and I like to use a different channel for each integration.  I don't have any specific suggestions since each site is different.

Extra context: 👍

Table name: I don't have a need to access to the tables directly, but I have experience removing malware from WP sites, and they always use randomly generated names like that.  When I first saw those names I thought my local site had somehow gotten infected.  Also, and this is very minor, I use phpmyadmin and those long table names get cut off in the sidebar.
